<h1 style="text-align: center;"><a title="Data Prep-AIMS-Cmr-2024-25">Assignment 1, Data Prep  @AIMSSen 25</h1>

**Instructor:** 

* Rockefeller

# Exercice 1:  Trends in Business in Nigeria

**NaijaBiz Analytics** is a leading business intelligence firm that specializes in analyzing small and medium enterprises (SMEs) across Nigeria. The company has been contracted by the Nigerian Ministry of Industry, Trade and Investment to conduct a comprehensive analysis of business performance across different states and sectors. 

As a newly hired Data Analyst at NaijaBiz Analytics, you have been provided with a dataset containing information about businesses across various Nigerian states. Your task is to analyze the business data using basic filtering operations to provide actionable insights.

**Dataset:** `naija_businesses.csv`

For each question below, provide the Python pandas code and extract meaningful business insights from your analysis.


### Importing pandas and loading data

In [1]:
import pandas as pd
df_NaijaBiz =  pd.read_csv( "data/naija_businesses.csv" )
df_NaijaBiz

,business_id,business_name,state,business_type,start_date,monthly_revenue_naira,monthly_costs_naira,employees,has_website,accepts_mobile_money
0,NG_0001,Business_1,Benue,Retail,2023-02-01,417741,406718,8,False,True
1,NG_0002,Business_2,Kaduna,Manufacturing,2023-11-17,696804,256593,10,True,False
2,NG_0003,Business_3,Oyo,Technology,2020-09-02,520713,87719,11,False,True
3,NG_0004,Business_4,Benue,Manufacturing,2023-04-14,553527,389310,7,False,True
4,NG_0005,Business_5,Rivers,Technology,2022-11-28,231537,-11869,10,True,False
...,...,...,...,...,...,...,...,...,...,...
195,NG_0196,Business_196,Kano,Services,2023-09-27,611128,415487,8,True,False
196,NG_0197,Business_197,Kano,Agriculture,2021-01-31,916895,349547,7,False,True
197,NG_0198,Business_198,Kaduna,Agriculture,2021-11-29,778180,430501,8,False,True
198,NG_0199,Business_199,Oyo,Agriculture,2022-11-08,98486,78788,4,False,True


### Q1: High Revenue Businesses
Filter and display all businesses that have monthly revenue greater than 600,000 Naira. How many such businesses exist?

**Business Context:** Identifying high-revenue businesses can help understand successful business models.


In [2]:
High_revenue_business=df_NaijaBiz[df_NaijaBiz['monthly_revenue_naira']>600000]
High_revenue_business

,business_id,business_name,state,business_type,start_date,monthly_revenue_naira,monthly_costs_naira,employees,has_website,accepts_mobile_money
1,NG_0002,Business_2,Kaduna,Manufacturing,2023-11-17,696804,256593,10,True,False
6,NG_0007,Business_7,Oyo,Services,2021-06-21,724708,207436,12,False,True
7,NG_0008,Business_8,Oyo,Technology,2022-07-09,928641,111173,11,False,False
11,NG_0012,Business_12,Benue,Services,2022-06-21,835058,436180,7,False,True
12,NG_0013,Business_13,Rivers,Manufacturing,2020-02-05,677796,298681,6,False,True
18,NG_0019,Business_19,Anambra,Manufacturing,2021-10-15,677240,486295,8,False,True
27,NG_0028,Business_28,Kano,Retail,2020-02-20,713895,168645,10,False,False
29,NG_0030,Business_30,Kaduna,Retail,2023-04-28,872155,371524,6,False,True
30,NG_0031,Business_31,Oyo,Services,2023-02-22,795650,40562,3,False,True
36,NG_0037,Business_37,Kaduna,Services,2021-11-12,788749,212612,10,False,True


In [3]:
High_revenue_business[['business_name','monthly_revenue_naira']]

,business_name,monthly_revenue_naira
1,Business_2,696804
6,Business_7,724708
7,Business_8,928641
11,Business_12,835058
12,Business_13,677796
18,Business_19,677240
27,Business_28,713895
29,Business_30,872155
30,Business_31,795650
36,Business_37,788749


In [4]:
df_NaijaBiz['monthly_revenue_naira'][df_NaijaBiz['monthly_revenue_naira']>600000].count()

np.int64(60)

### Comment

About 30% of the businesses are making high revenue. This shows the data focuses mostly on successful, established companies rather than very small, struggling ones.

### Q2: Technology Businesses with Websites
Filter businesses that are in the "Technology" sector AND have a website. Display their business names and monthly revenue.

**Business Context:** Understanding digital adoption in the tech sector can inform technology support programs.


In [5]:
Technology_Website=df_NaijaBiz[(df_NaijaBiz['business_type']=='Technology') &(df_NaijaBiz['has_website']==True)]
Technology_Website

,business_id,business_name,state,business_type,start_date,monthly_revenue_naira,monthly_costs_naira,employees,has_website,accepts_mobile_money
4,NG_0005,Business_5,Rivers,Technology,2022-11-28,231537,-11869,10,True,False
9,NG_0010,Business_10,Kano,Technology,2021-06-08,549862,293803,3,True,False
33,NG_0034,Business_34,Kano,Technology,2021-07-23,265604,8259,8,True,True
74,NG_0075,Business_75,Oyo,Technology,2022-12-17,370740,296592,6,True,True
115,NG_0116,Business_116,Kaduna,Technology,2020-12-22,742563,272479,8,True,True
144,NG_0145,Business_145,Anambra,Technology,2022-12-02,349510,294444,8,True,True
161,NG_0162,Business_162,Lagos,Technology,2020-04-24,319976,255980,12,True,False
176,NG_0177,Business_177,Rivers,Technology,2022-07-08,673369,261477,10,True,False


In [6]:

Technology_Website[['business_name','monthly_revenue_naira']]

,business_name,monthly_revenue_naira
4,Business_5,231537
9,Business_10,549862
33,Business_34,265604
74,Business_75,370740
115,Business_116,742563
144,Business_145,349510
161,Business_162,319976
176,Business_177,673369


### Comment
Only 8 technology businesses have a website. This is strange. It suggests these are likely small repair shops or phone stalls, not software companies, since software companies usually have websites.

### Q3: Manufacturing Businesses in Specific States
Filter all Manufacturing businesses located in either "Lagos" OR "Kano" states. How many manufacturing businesses are in these two states?

**Business Context:** These states are major industrial hubs, and this analysis helps understand manufacturing concentration.


In [7]:
Manufacturing_businesses=df_NaijaBiz[(df_NaijaBiz['state']=='Lagos') | (df_NaijaBiz['state']=='Kano')]
Manufacturing_businesses_=Manufacturing_businesses[['state','business_type']]
Manufacturing_businesses_[Manufacturing_businesses['business_type']=='Manufacturing']

,state,business_type
22,Kano,Manufacturing
42,Kano,Manufacturing
70,Kano,Manufacturing
75,Kano,Manufacturing
87,Lagos,Manufacturing
92,Kano,Manufacturing
105,Kano,Manufacturing
122,Kano,Manufacturing
127,Lagos,Manufacturing
165,Lagos,Manufacturing


In [8]:
Manufacturing_businesses_[Manufacturing_businesses['business_type']=='Manufacturing'].count()

state            10
business_type    10
dtype: int64

### Comment
These two countries make up almost 40% of your data. If you build a business plan based on this, it might not work for other places like West Africa because the data is skewed.

### Q4: Small Businesses Analysis
Filter businesses that have fewer than 5 employees AND monthly costs less than 200,000 Naira. What business types do these small businesses represent?

**Business Context:** Small businesses often need different support strategies than larger enterprises.


In [9]:
small_business=df_NaijaBiz[(df_NaijaBiz['employees']<=5) &(df_NaijaBiz['monthly_costs_naira']<=200000)]
small_business

,business_id,business_name,state,business_type,start_date,monthly_revenue_naira,monthly_costs_naira,employees,has_website,accepts_mobile_money
5,NG_0006,Business_6,Anambra,Technology,2022-01-28,-172819,-138255,5,False,False
16,NG_0017,Business_17,Kaduna,Retail,2021-05-26,212107,169685,2,True,False
23,NG_0024,Business_24,Anambra,Services,2023-10-09,207648,166118,4,False,True
30,NG_0031,Business_31,Oyo,Services,2023-02-22,795650,40562,3,False,True
34,NG_0035,Business_35,Imo,Retail,2023-09-27,182707,89967,5,True,True
53,NG_0054,Business_54,Kaduna,Retail,2023-10-05,479231,197888,5,False,False
57,NG_0058,Business_58,Lagos,Services,2022-07-24,306577,43908,4,True,True
71,NG_0072,Business_72,Kano,Services,2023-02-24,167976,70428,1,False,True
83,NG_0084,Business_84,Anambra,Services,2022-06-15,805420,163626,4,False,True
89,NG_0090,Business_90,Kano,Agriculture,2022-05-10,403124,-29477,4,False,True


In [10]:
small_business['business_type']


5         Technology
16            Retail
23          Services
30          Services
34            Retail
53            Retail
57          Services
71          Services
83          Services
89       Agriculture
122    Manufacturing
123           Retail
138       Technology
153         Services
156      Agriculture
178           Retail
181    Manufacturing
198      Agriculture
Name: business_type, dtype: object

### Comment
Small businesses exist in every sector, including Manufacturing and Tech. This proves you don't need to be a big company to be in these industries.

### Q5: Profitable Retail Businesses
Filter Retail businesses where monthly revenue is greater than monthly costs. Display their state and employee count.

**Business Context:** Understanding profitable retail operations can help identify successful business practices.


In [11]:
Retail_Profit=df_NaijaBiz[
    (df_NaijaBiz['business_type']=='Retail') &
    (df_NaijaBiz['monthly_revenue_naira']>df_NaijaBiz['monthly_costs_naira'])]
Retail_Profit

,business_id,business_name,state,business_type,start_date,monthly_revenue_naira,monthly_costs_naira,employees,has_website,accepts_mobile_money
0,NG_0001,Business_1,Benue,Retail,2023-02-01,417741,406718,8,False,True
16,NG_0017,Business_17,Kaduna,Retail,2021-05-26,212107,169685,2,True,False
26,NG_0027,Business_27,Imo,Retail,2022-05-03,151342,121073,12,False,True
27,NG_0028,Business_28,Kano,Retail,2020-02-20,713895,168645,10,False,False
28,NG_0029,Business_29,Anambra,Retail,2021-07-28,467226,270305,3,True,True
29,NG_0030,Business_30,Kaduna,Retail,2023-04-28,872155,371524,6,False,True
31,NG_0032,Business_32,Lagos,Retail,2020-05-29,324106,259284,3,False,True
34,NG_0035,Business_35,Imo,Retail,2023-09-27,182707,89967,5,True,True
37,NG_0038,Business_38,Lagos,Retail,2022-04-17,117018,93614,10,True,True
39,NG_0040,Business_40,Rivers,Retail,2022-09-29,653667,146427,6,False,False


In [12]:
Retail_Profit[['state','employees']].count()

state        50
employees    50
dtype: int64

In [13]:
Retail_Profit['employees'].value_counts()

employees
10    11
8      9
5      6
6      5
9      5
11     4
3      3
2      2
7      2
12     1
13     1
14     1
Name: count, dtype: int64

### Comment
Making a profit isn't about how many people you hire. Tiny shops with 2 employees are profitable, and larger ones with 14 employees are too. Size doesn't guarantee success here.

### Q6: Businesses Accepting Mobile Money
Filter all businesses that accept mobile money AND have more than 8 employees. Which states are these businesses located in?

**Business Context:** Larger businesses adopting mobile payments indicate financial technology integration.


In [15]:
Business_mobile_money= df_NaijaBiz[(df_NaijaBiz['employees'] >= 8 ) & (df_NaijaBiz['accepts_mobile_money']== True)]
Business_mobile_money


,business_id,business_name,state,business_type,start_date,monthly_revenue_naira,monthly_costs_naira,employees,has_website,accepts_mobile_money
0,NG_0001,Business_1,Benue,Retail,2023-02-01,417741,406718,8,False,True
2,NG_0003,Business_3,Oyo,Technology,2020-09-02,520713,87719,11,False,True
6,NG_0007,Business_7,Oyo,Services,2021-06-21,724708,207436,12,False,True
10,NG_0011,Business_11,Rivers,Services,2022-04-07,564903,326094,16,False,True
13,NG_0014,Business_14,Rivers,Agriculture,2021-11-15,599212,367711,13,False,True
...,...,...,...,...,...,...,...,...,...,...
189,NG_0190,Business_190,Imo,Retail,2022-05-16,261347,209077,8,False,True
191,NG_0192,Business_192,Rivers,Agriculture,2023-07-20,421793,141370,11,True,True
193,NG_0194,Business_194,Benue,Agriculture,2021-08-18,934568,261431,9,True,True
194,NG_0195,Business_195,Benue,Agriculture,2021-12-29,812562,504472,11,False,True


In [16]:
Business_mobile_money['state'].unique()

array(['Benue', 'Oyo', 'Rivers', 'Anambra', 'Imo', 'Kano', 'Kaduna',
       'Lagos'], dtype=object)

### Comment

Big companies are using mobile money a lot. This means mobile money isn't just for friends sending cash to friends; it's a real tool for serious business payments now.

### Q7: Low-Cost Operations
Filter businesses where monthly costs are less than 150,000 Naira. What is the most common business type among these low-cost operations?

**Business Context:** Low-cost business models can be important for economic inclusion and job creation.


In [17]:
df_NaijaBiz[
    (df_NaijaBiz['monthly_costs_naira']<=150000)]

,business_id,business_name,state,business_type,start_date,monthly_revenue_naira,monthly_costs_naira,employees,has_website,accepts_mobile_money
2,NG_0003,Business_3,Oyo,Technology,2020-09-02,520713,87719,11,False,True
4,NG_0005,Business_5,Rivers,Technology,2022-11-28,231537,-11869,10,True,False
5,NG_0006,Business_6,Anambra,Technology,2022-01-28,-172819,-138255,5,False,False
7,NG_0008,Business_8,Oyo,Technology,2022-07-09,928641,111173,11,False,False
26,NG_0027,Business_27,Imo,Retail,2022-05-03,151342,121073,12,False,True
30,NG_0031,Business_31,Oyo,Services,2023-02-22,795650,40562,3,False,True
33,NG_0034,Business_34,Kano,Technology,2021-07-23,265604,8259,8,True,True
34,NG_0035,Business_35,Imo,Retail,2023-09-27,182707,89967,5,True,True
37,NG_0038,Business_38,Lagos,Retail,2022-04-17,117018,93614,10,True,True
39,NG_0040,Business_40,Rivers,Retail,2022-09-29,653667,146427,6,False,False


In [18]:
low_cost=df_NaijaBiz[(df_NaijaBiz['monthly_costs_naira']<=150000)]
low_cost

,business_id,business_name,state,business_type,start_date,monthly_revenue_naira,monthly_costs_naira,employees,has_website,accepts_mobile_money
2,NG_0003,Business_3,Oyo,Technology,2020-09-02,520713,87719,11,False,True
4,NG_0005,Business_5,Rivers,Technology,2022-11-28,231537,-11869,10,True,False
5,NG_0006,Business_6,Anambra,Technology,2022-01-28,-172819,-138255,5,False,False
7,NG_0008,Business_8,Oyo,Technology,2022-07-09,928641,111173,11,False,False
26,NG_0027,Business_27,Imo,Retail,2022-05-03,151342,121073,12,False,True
30,NG_0031,Business_31,Oyo,Services,2023-02-22,795650,40562,3,False,True
33,NG_0034,Business_34,Kano,Technology,2021-07-23,265604,8259,8,True,True
34,NG_0035,Business_35,Imo,Retail,2023-09-27,182707,89967,5,True,True
37,NG_0038,Business_38,Lagos,Retail,2022-04-17,117018,93614,10,True,True
39,NG_0040,Business_40,Rivers,Retail,2022-09-29,653667,146427,6,False,False


In [19]:
low_cost['business_type'].mode()[0]

'Retail'

### Q8: Recent Business Registrations
Filter businesses that started in 2023. Which business type is most common among these new businesses?

**Business Context:** Recent registration trends can indicate emerging business opportunities and economic growth sectors.


In [20]:
df_NaijaBiz['start_date'] = pd.to_datetime(df_NaijaBiz['start_date'])
recent_businesses = df_NaijaBiz[df_NaijaBiz['start_date'].dt.year == 2023]
recent_businesses

,business_id,business_name,state,business_type,start_date,monthly_revenue_naira,monthly_costs_naira,employees,has_website,accepts_mobile_money
0,NG_0001,Business_1,Benue,Retail,2023-02-01,417741,406718,8,False,True
1,NG_0002,Business_2,Kaduna,Manufacturing,2023-11-17,696804,256593,10,True,False
3,NG_0004,Business_4,Benue,Manufacturing,2023-04-14,553527,389310,7,False,True
21,NG_0022,Business_22,Oyo,Manufacturing,2023-03-12,264774,211819,9,True,True
22,NG_0023,Business_23,Kano,Manufacturing,2023-01-20,589101,332379,11,False,False
23,NG_0024,Business_24,Anambra,Services,2023-10-09,207648,166118,4,False,True
29,NG_0030,Business_30,Kaduna,Retail,2023-04-28,872155,371524,6,False,True
30,NG_0031,Business_31,Oyo,Services,2023-02-22,795650,40562,3,False,True
34,NG_0035,Business_35,Imo,Retail,2023-09-27,182707,89967,5,True,True
47,NG_0048,Business_48,Benue,Retail,2023-01-31,331178,264942,2,False,True


In [21]:
recent_businesses['business_type'].mode()[0]

'Retail'

### Q9: Large Employers
Filter businesses that have 10 or more employees. Calculate the average monthly revenue of these larger employers.

**Business Context:** Large employers are important for job creation and economic stability.


In [22]:
large_business=df_NaijaBiz[df_NaijaBiz['employees'] > 10]
large_business

,business_id,business_name,state,business_type,start_date,monthly_revenue_naira,monthly_costs_naira,employees,has_website,accepts_mobile_money
2,NG_0003,Business_3,Oyo,Technology,2020-09-02,520713,87719,11,False,True
6,NG_0007,Business_7,Oyo,Services,2021-06-21,724708,207436,12,False,True
7,NG_0008,Business_8,Oyo,Technology,2022-07-09,928641,111173,11,False,False
10,NG_0011,Business_11,Rivers,Services,2022-04-07,564903,326094,16,False,True
13,NG_0014,Business_14,Rivers,Agriculture,2021-11-15,599212,367711,13,False,True
19,NG_0020,Business_20,Rivers,Manufacturing,2022-08-29,453259,287135,12,False,False
22,NG_0023,Business_23,Kano,Manufacturing,2023-01-20,589101,332379,11,False,False
26,NG_0027,Business_27,Imo,Retail,2022-05-03,151342,121073,12,False,True
32,NG_0033,Business_33,Kaduna,Manufacturing,2020-11-09,341656,273324,11,True,False
41,NG_0042,Business_42,Benue,Agriculture,2021-09-04,646634,369677,14,False,True


In [23]:
large_business['monthly_revenue_naira'].mean()

np.float64(502536.35294117645)

### Comment
Big companies with more staff actually make less money on average (502k) than the "High Revenue" cutoff (600k). This means hiring more people isn't automatically helping them make more money.

### Q10: Digital-Ready Businesses
Filter businesses that have a website AND accept mobile money. What percentage of the total businesses are digitally ready?

**Business Context:** Digital readiness indicates how prepared businesses are for the modern economy.


In [24]:
Digital_ready_businesses=df_NaijaBiz[(df_NaijaBiz['accepts_mobile_money']==True) &(df_NaijaBiz['has_website']==True)]
Digital_ready_businesses

,business_id,business_name,state,business_type,start_date,monthly_revenue_naira,monthly_costs_naira,employees,has_website,accepts_mobile_money
14,NG_0015,Business_15,Anambra,Agriculture,2020-01-20,511114,422377,8,True,True
21,NG_0022,Business_22,Oyo,Manufacturing,2023-03-12,264774,211819,9,True,True
28,NG_0029,Business_29,Anambra,Retail,2021-07-28,467226,270305,3,True,True
33,NG_0034,Business_34,Kano,Technology,2021-07-23,265604,8259,8,True,True
34,NG_0035,Business_35,Imo,Retail,2023-09-27,182707,89967,5,True,True
37,NG_0038,Business_38,Lagos,Retail,2022-04-17,117018,93614,10,True,True
40,NG_0041,Business_41,Rivers,Services,2021-03-19,610938,493260,8,True,True
45,NG_0046,Business_46,Kaduna,Manufacturing,2022-09-30,8161,6528,16,True,True
56,NG_0057,Business_57,Anambra,Retail,2022-04-02,359637,287709,9,True,True
57,NG_0058,Business_58,Lagos,Services,2022-07-24,306577,43908,4,True,True


In [25]:

Digital_businesses= Digital_ready_businesses.shape[0]
Total_businesses = df_NaijaBiz.shape[0]
Percentage = (Digital_businesses/ Total_businesses) * 100
Percentage


14.000000000000002

### Comment
Only 14% of businesses are fully digital. Since we know many use mobile money, the real problem is that they don't have websites. To help them, we should focus on getting them online.

# Exercise 2: African Agriculture Analysis

**AgriTech Africa** is a research organization analyzing agricultural technology adoption across five African countries. You have been provided with a dataset containing information about farming communities and their technology usage patterns.

For each question below, provide the Python pandas code and provide a key insight you can get from your answer.


In [26]:
import pandas as pd
Agri_tech =  pd.read_csv( "data/african_agriculture_dataset.csv" )
Agri_tech

,Community_Name,Country,Crop_Type,Farm_Size,Tech_Adoption_Index,Drone_Usage,Mobile_Tech,Annual_Yield,Fertilizer_Type,Irrigation,Weather_Station,Annual_Rainfall
0,Mountain Cultivators Club,South Africa,Barley,39.96,40,Yes,SMS,318.05,Organic,Sprinkler,Yes,1046.21
1,Prairie Cultivators Club,South Africa,Sorghum,77.62,48,No,SMS,704.85,Organic,Drip,No,921.82
2,Prairie Agri Society,Nigeria,Wheat,97.46,14,Yes,SMS,439.41,NaN,Drip,Yes,1072.58
3,Harvest Farmers Union,Nigeria,Maize,27.84,55,No,SMS,668.23,NaN,Sprinkler,No,629.73
4,Valley Farmers Group,Ghana,Sorghum,70.60,53,Yes,Mobile Apps,801.16,NaN,Sprinkler,No,909.69
...,...,...,...,...,...,...,...,...,...,...,...,...
1995,Harvest Agri Society,Kenya,Maize,39.93,0,Yes,NaN,246.61,NaN,NaN,Yes,1123.58
1996,Mountain Farmers Group,Nigeria,Maize,50.99,90,No,NaN,205.38,NaN,Drip,Yes,646.68
1997,Prairie Cultivators Club,Ghana,Rice,89.37,43,No,Mobile Apps,603.02,Organic,Drip,No,958.79
1998,Sunrise Agricultural Association,Kenya,Wheat,87.01,99,No,Mobile Apps,370.42,Inorganic,Sprinkler,Yes,677.78


## Q1: Identifying Technology Leaders
**Business Question:** Which farming communities are leading in technology adoption, and how many of our partner communities have achieved a Technology Adoption Index above 80?

**Context:** Understanding technology leaders helps identify best practices for scaling across other communities.

In [27]:
Technology_leaders=Agri_tech[(Agri_tech['Tech_Adoption_Index']>=80)]
Technology_leaders

,Community_Name,Country,Crop_Type,Farm_Size,Tech_Adoption_Index,Drone_Usage,Mobile_Tech,Annual_Yield,Fertilizer_Type,Irrigation,Weather_Station,Annual_Rainfall
8,Prairie Agricultural Association,Tanzania,Wheat,23.15,86,Yes,SMS,504.69,Organic,Drip,Yes,621.79
11,Sunrise Growers Coalition,South Africa,Wheat,70.31,85,Yes,SMS,409.37,Organic,Sprinkler,No,1178.04
16,River Agri Society,Tanzania,Sorghum,74.55,83,No,Mobile Apps,691.11,Inorganic,Sprinkler,Yes,719.05
17,Savannah Agri Society,South Africa,Sorghum,14.50,91,No,Mobile Apps,398.23,Inorganic,Drip,Yes,1067.74
20,Savannah Cultivators Club,Kenya,Sorghum,46.27,92,No,SMS,686.81,NaN,NaN,Yes,612.64
...,...,...,...,...,...,...,...,...,...,...,...,...
1991,Meadow Farmers Group,Kenya,Rice,84.23,91,Yes,SMS,273.06,Inorganic,Sprinkler,No,658.16
1993,Golden Agri Society,Kenya,Rice,90.02,80,Yes,Mobile Apps,664.49,NaN,Drip,Yes,710.92
1994,Green Cultivators Club,Nigeria,Wheat,26.81,95,No,SMS,454.72,NaN,Sprinkler,Yes,607.55
1996,Mountain Farmers Group,Nigeria,Maize,50.99,90,No,NaN,205.38,NaN,Drip,Yes,646.68


In [28]:
Technology_leaders['Community_Name'].count()

np.int64(409)

In [29]:
Technology_leaders[['Community_Name','Tech_Adoption_Index']]

,Community_Name,Tech_Adoption_Index
8,Prairie Agricultural Association,86
11,Sunrise Growers Coalition,85
16,River Agri Society,83
17,Savannah Agri Society,91
20,Savannah Cultivators Club,92
...,...,...
1991,Meadow Farmers Group,91
1993,Golden Agri Society,80
1994,Green Cultivators Club,95
1996,Mountain Farmers Group,90


In [30]:
Technology_leaders['Community_Name'].mode()[0]

'Harvest Farmers Union'

### Comment
A massive 20% of the dataset (409 communities) are tech leaders. A scan of the Community_Name column (e.g., "Growers Coalition," "Farmers Union") suggests that high tech adoption is driven by cooperative structures. Individual farmers aren't buying this tech; groups are pooling resources to acquire it.

## Q2: Premium Drone Adoption Analysis
**Business Question:** Among large-scale farming operations (over 70 acres), which communities have invested in drone technology, and where are they located?

**Context:** Large farms with drone adoption represent high-value clients for advanced agricultural technology partnerships.

In [31]:
large_scale=Agri_tech[(Agri_tech['Drone_Usage']=='Yes') &(Agri_tech['Farm_Size']>=70)]
large_scale

,Community_Name,Country,Crop_Type,Farm_Size,Tech_Adoption_Index,Drone_Usage,Mobile_Tech,Annual_Yield,Fertilizer_Type,Irrigation,Weather_Station,Annual_Rainfall
2,Prairie Agri Society,Nigeria,Wheat,97.46,14,Yes,SMS,439.41,NaN,Drip,Yes,1072.58
4,Valley Farmers Group,Ghana,Sorghum,70.60,53,Yes,Mobile Apps,801.16,NaN,Sprinkler,No,909.69
5,Savannah Agri Society,Tanzania,Barley,92.92,30,Yes,SMS,784.90,Organic,Sprinkler,No,872.48
11,Sunrise Growers Coalition,South Africa,Wheat,70.31,85,Yes,SMS,409.37,Organic,Sprinkler,No,1178.04
24,Meadow Farmers Group,Ghana,Rice,94.51,51,Yes,NaN,883.63,NaN,NaN,No,917.49
...,...,...,...,...,...,...,...,...,...,...,...,...
1968,Harvest Growers Coalition,Tanzania,Barley,95.20,87,Yes,Mobile Apps,487.51,NaN,Sprinkler,Yes,583.40
1974,Meadow Growers Coalition,Kenya,Barley,77.23,79,Yes,Mobile Apps,425.77,NaN,NaN,Yes,422.86
1988,Savannah Cultivators Club,Ghana,Wheat,97.13,29,Yes,Mobile Apps,555.20,Inorganic,Sprinkler,No,663.25
1991,Meadow Farmers Group,Kenya,Rice,84.23,91,Yes,SMS,273.06,Inorganic,Sprinkler,No,658.16


In [32]:
Agri_tech['Farm_Size']>70

0       False
1        True
2        True
3       False
4        True
        ...  
1995    False
1996    False
1997     True
1998     True
1999    False
Name: Farm_Size, Length: 2000, dtype: bool

In [33]:
large_scale['Country'].unique()

array(['Nigeria', 'Ghana', 'Tanzania', 'South Africa', 'Kenya'],
      dtype=object)

### Comment
Drone usage (338 communities) is almost exclusively tied to large-scale operations. This confirms that drones are currently a tool for industrial monoculture (monitoring vast acreages of Wheat/Barley) and have not yet been adapted for the complex, inter-cropped plots typical of smallholder farming.

## Q3: Market Expansion Opportunities
**Business Question:** How many farming communities are operating in our priority markets of South Africa and Kenya?

**Context:** These countries represent key expansion opportunities for agricultural technology solutions.

In [34]:
countries=['Kenya','South Africa']

In [35]:
Market_expansion=Agri_tech[Agri_tech['Country'].isin(countries)]
Market_expansion

,Community_Name,Country,Crop_Type,Farm_Size,Tech_Adoption_Index,Drone_Usage,Mobile_Tech,Annual_Yield,Fertilizer_Type,Irrigation,Weather_Station,Annual_Rainfall
0,Mountain Cultivators Club,South Africa,Barley,39.96,40,Yes,SMS,318.05,Organic,Sprinkler,Yes,1046.21
1,Prairie Cultivators Club,South Africa,Sorghum,77.62,48,No,SMS,704.85,Organic,Drip,No,921.82
6,River Growers Coalition,South Africa,Sorghum,68.94,60,No,NaN,437.78,Inorganic,NaN,Yes,897.94
9,Sunrise Growers Coalition,Kenya,Maize,53.98,59,Yes,NaN,883.72,Organic,NaN,No,629.33
11,Sunrise Growers Coalition,South Africa,Wheat,70.31,85,Yes,SMS,409.37,Organic,Sprinkler,No,1178.04
...,...,...,...,...,...,...,...,...,...,...,...,...
1989,Harvest Agri Society,Kenya,Rice,55.40,24,No,SMS,658.68,NaN,Sprinkler,Yes,693.57
1991,Meadow Farmers Group,Kenya,Rice,84.23,91,Yes,SMS,273.06,Inorganic,Sprinkler,No,658.16
1993,Golden Agri Society,Kenya,Rice,90.02,80,Yes,Mobile Apps,664.49,NaN,Drip,Yes,710.92
1995,Harvest Agri Society,Kenya,Maize,39.93,0,Yes,NaN,246.61,NaN,NaN,Yes,1123.58


In [36]:
Market_expansion['Community_Name'].count()

np.int64(760)

### Comment
These two markets constitute nearly 40% of the total dataset (760 rows). Any predictive model built on this data will be heavily biased toward the East/Southern African context and may not generalize well to West Africa (Nigeria/Ghana) without re-balancing.

## Q4: Underserved Market Identification
**Business Question:** Which countries have small-scale, low-technology farming operations (under 20 acres with Technology Adoption Index below 30) that could benefit from basic technology interventions?

**Context:** Small, low-tech farms represent an underserved market segment requiring tailored entry-level solutions.


In [37]:
underserved_market=Agri_tech[(Agri_tech['Tech_Adoption_Index']<=20) &(Agri_tech['Farm_Size']<=30)]
underserved_market

,Community_Name,Country,Crop_Type,Farm_Size,Tech_Adoption_Index,Drone_Usage,Mobile_Tech,Annual_Yield,Fertilizer_Type,Irrigation,Weather_Station,Annual_Rainfall
21,Harvest Farmers Group,Ghana,Wheat,22.35,4,Yes,Mobile Apps,242.71,Inorganic,Drip,No,576.68
54,Sunrise Agricultural Association,South Africa,Wheat,3.14,10,No,NaN,531.49,Inorganic,Sprinkler,Yes,405.82
81,Mountain Growers Coalition,Kenya,Rice,13.85,1,No,NaN,801.18,NaN,Sprinkler,No,1007.81
106,Green Farmers Union,Nigeria,Sorghum,13.59,13,Yes,NaN,697.87,Organic,Sprinkler,No,767.65
111,Mountain Farmers Group,South Africa,Sorghum,28.15,16,No,Mobile Apps,822.56,NaN,NaN,Yes,435.01
...,...,...,...,...,...,...,...,...,...,...,...,...
1919,Mountain Cultivators Club,Tanzania,Sorghum,15.37,17,No,SMS,786.98,NaN,Sprinkler,Yes,581.72
1955,Prairie Agricultural Association,South Africa,Sorghum,26.18,7,Yes,Mobile Apps,799.44,Inorganic,Sprinkler,Yes,562.10
1976,Mountain Cultivators Club,Ghana,Wheat,18.76,16,No,Mobile Apps,781.23,Inorganic,Drip,Yes,463.61
1978,Meadow Agri Society,Kenya,Wheat,2.19,2,Yes,NaN,112.96,Inorganic,NaN,Yes,643.36


In [38]:
underserved_market['Country'].unique()

array(['Ghana', 'South Africa', 'Kenya', 'Nigeria', 'Tanzania'],
      dtype=object)

### Comment
Only 123 communities (6%) fit the "Underserved" profile. In reality, this segment represents the majority of African agriculture. The low count here implies the dataset itself is biased toward organized, successful farming groups and under-represents the true subsistence baseline.

## Q5: Sustainable Agriculture Success Stories
**Business Question:** Which organic farming communities are achieving high productivity (over 600 annual yield), and what irrigation technologies are they using?

**Context:** Successful organic farms demonstrate sustainable practices that can be promoted to environmentally conscious markets.

In [39]:
Sustaintable_Agri=Agri_tech[(Agri_tech['Annual_Yield']>=600)]
Sustaintable_Agri

,Community_Name,Country,Crop_Type,Farm_Size,Tech_Adoption_Index,Drone_Usage,Mobile_Tech,Annual_Yield,Fertilizer_Type,Irrigation,Weather_Station,Annual_Rainfall
1,Prairie Cultivators Club,South Africa,Sorghum,77.62,48,No,SMS,704.85,Organic,Drip,No,921.82
3,Harvest Farmers Union,Nigeria,Maize,27.84,55,No,SMS,668.23,NaN,Sprinkler,No,629.73
4,Valley Farmers Group,Ghana,Sorghum,70.60,53,Yes,Mobile Apps,801.16,NaN,Sprinkler,No,909.69
5,Savannah Agri Society,Tanzania,Barley,92.92,30,Yes,SMS,784.90,Organic,Sprinkler,No,872.48
7,Savannah Farmers Union,Nigeria,Rice,68.43,57,No,NaN,611.11,Organic,Drip,No,534.91
...,...,...,...,...,...,...,...,...,...,...,...,...
1989,Harvest Agri Society,Kenya,Rice,55.40,24,No,SMS,658.68,NaN,Sprinkler,Yes,693.57
1990,Golden Farmers Union,Nigeria,Wheat,19.07,6,No,SMS,904.24,NaN,NaN,No,736.37
1993,Golden Agri Society,Kenya,Rice,90.02,80,Yes,Mobile Apps,664.49,NaN,Drip,Yes,710.92
1997,Prairie Cultivators Club,Ghana,Rice,89.37,43,No,Mobile Apps,603.02,Organic,Drip,No,958.79


In [40]:
Sustaintable_Agri[['Community_Name','Irrigation']]

,Community_Name,Irrigation
1,Prairie Cultivators Club,Drip
3,Harvest Farmers Union,Sprinkler
4,Valley Farmers Group,Sprinkler
5,Savannah Agri Society,Sprinkler
7,Savannah Farmers Union,Drip
...,...,...
1989,Harvest Agri Society,Sprinkler
1990,Golden Farmers Union,NaN
1993,Golden Agri Society,Drip
1997,Prairie Cultivators Club,Drip


In [41]:
Sustaintable_Agri['Irrigation'].unique()

array(['Drip', 'Sprinkler', nan], dtype=object)

### Comment
A massive 863 communities achieve high yields, but looking at the Irrigation column, both "Drip" and "Sprinkler" are heavily represented. This suggests that Water Management is the primary driver of high yield, regardless of whether the farm uses Organic or Inorganic fertilizer.

## Q6: Climate-Smart Agriculture Adoption
**Business Question:** In high-rainfall regions (over 1000mm annually), which countries have farming communities that have invested in weather monitoring technology?

**Context:** Weather station adoption in high-rainfall areas indicates sophisticated climate risk management practices.


In [42]:
Climate_smart=Agri_tech[(Agri_tech['Annual_Rainfall']>=1000) &(Agri_tech['Weather_Station']=='Yes')]
Climate_smart

,Community_Name,Country,Crop_Type,Farm_Size,Tech_Adoption_Index,Drone_Usage,Mobile_Tech,Annual_Yield,Fertilizer_Type,Irrigation,Weather_Station,Annual_Rainfall
0,Mountain Cultivators Club,South Africa,Barley,39.96,40,Yes,SMS,318.05,Organic,Sprinkler,Yes,1046.21
2,Prairie Agri Society,Nigeria,Wheat,97.46,14,Yes,SMS,439.41,NaN,Drip,Yes,1072.58
12,Valley Agricultural Association,Ghana,Barley,99.69,60,No,Mobile Apps,182.56,Organic,Drip,Yes,1163.25
17,Savannah Agri Society,South Africa,Sorghum,14.50,91,No,Mobile Apps,398.23,Inorganic,Drip,Yes,1067.74
18,Mountain Agri Society,Nigeria,Wheat,62.62,26,Yes,NaN,700.75,NaN,Sprinkler,Yes,1132.60
...,...,...,...,...,...,...,...,...,...,...,...,...
1937,Harvest Agricultural Association,Tanzania,Sorghum,38.29,41,No,NaN,381.90,Inorganic,NaN,Yes,1042.26
1938,Prairie Growers Coalition,South Africa,Rice,58.13,41,No,SMS,291.14,Organic,Drip,Yes,1045.60
1970,River Cultivators Club,Ghana,Wheat,54.14,2,No,Mobile Apps,826.23,NaN,NaN,Yes,1176.19
1987,Green Agri Society,Kenya,Barley,20.11,39,Yes,SMS,431.03,Organic,Drip,Yes,1081.81


In [43]:
Climate_smart[['Country','Community_Name']]

,Country,Community_Name
0,South Africa,Mountain Cultivators Club
2,Nigeria,Prairie Agri Society
12,Ghana,Valley Agricultural Association
17,South Africa,Savannah Agri Society
18,Nigeria,Mountain Agri Society
...,...,...
1937,Tanzania,Harvest Agricultural Association
1938,South Africa,Prairie Growers Coalition
1970,Ghana,River Cultivators Club
1987,Kenya,Green Agri Society


### Comment

Weather stations are clustered in high-rainfall zones (238 counts). This indicates a flood-risk management strategy (monitoring excess rain) rather than a drought-risk strategy. Farmers are investing in data where the weather is volatile/heavy, likely to protect high-value cash crops from wash-out or fungal diseases.

## Q7: Entry-Level Market Assessment
**Business Question:** Among low-performing farming operations (annual yield under 200), which crop types are most prevalent and may need immediate intervention?

**Context:** Low-yield operations represent potential clients for productivity improvement solutions.

In [44]:
Level_Assessment=Agri_tech[Agri_tech['Annual_Yield']<=200]
Level_Assessment

,Community_Name,Country,Crop_Type,Farm_Size,Tech_Adoption_Index,Drone_Usage,Mobile_Tech,Annual_Yield,Fertilizer_Type,Irrigation,Weather_Station,Annual_Rainfall
12,Valley Agricultural Association,Ghana,Barley,99.69,60,No,Mobile Apps,182.56,Organic,Drip,Yes,1163.25
30,Harvest Farmers Union,Nigeria,Sorghum,5.88,83,Yes,Mobile Apps,136.35,NaN,Drip,Yes,1024.79
39,Meadow Agricultural Association,Ghana,Rice,16.93,71,No,NaN,181.94,Organic,NaN,Yes,1052.89
44,River Cultivators Club,Ghana,Barley,31.74,46,Yes,Mobile Apps,198.41,Organic,Drip,Yes,498.94
55,Mountain Agri Society,Nigeria,Wheat,59.81,52,No,SMS,154.01,Organic,Drip,Yes,1137.86
...,...,...,...,...,...,...,...,...,...,...,...,...
1950,Valley Agricultural Association,Tanzania,Wheat,43.91,9,No,SMS,150.15,Inorganic,Drip,Yes,519.95
1965,Mountain Farmers Group,Tanzania,Maize,22.47,88,Yes,SMS,139.24,Organic,Sprinkler,No,1100.21
1966,Valley Cultivators Club,Ghana,Rice,86.62,96,No,SMS,184.45,NaN,NaN,No,483.89
1978,Meadow Agri Society,Kenya,Wheat,2.19,2,Yes,NaN,112.96,Inorganic,NaN,Yes,643.36


In [45]:
Level_Assessment['Crop_Type'].unique()

array(['Barley', 'Sorghum', 'Rice', 'Wheat', 'Maize'], dtype=object)

### Comment
Crucially, some of these low-yield farms are using Drones and Mobile Apps (e.g., Row 44, Row 1992). This reveals a "Productivity Paradox"—technology is present, but it isn't fixing the yield. This implies the root cause of failure isn't lack of tech, but likely fundamental issues like Soil Health or Seed Quality that apps/drones cannot fix.

## Q8: Digital Agriculture Penetration
**Business Question:** What irrigation methods are most commonly used by digitally-enabled farming communities (those using Mobile Apps or SMS technology)?

**Context:** Understanding technology adoption patterns helps design integrated digital agriculture solutions.

In [46]:
Digital_Agri_Pen=Agri_tech[(Agri_tech['Mobile_Tech']=='Mobile Apps') |(Agri_tech['Mobile_Tech']=='Sms')]
Digital_Agri_Pen

,Community_Name,Country,Crop_Type,Farm_Size,Tech_Adoption_Index,Drone_Usage,Mobile_Tech,Annual_Yield,Fertilizer_Type,Irrigation,Weather_Station,Annual_Rainfall
4,Valley Farmers Group,Ghana,Sorghum,70.60,53,Yes,Mobile Apps,801.16,NaN,Sprinkler,No,909.69
12,Valley Agricultural Association,Ghana,Barley,99.69,60,No,Mobile Apps,182.56,Organic,Drip,Yes,1163.25
13,Mountain Agri Society,South Africa,Sorghum,4.33,79,Yes,Mobile Apps,250.82,Organic,NaN,No,581.75
14,Golden Farmers Union,Nigeria,Barley,7.46,41,Yes,Mobile Apps,430.30,Inorganic,Sprinkler,Yes,643.88
16,River Agri Society,Tanzania,Sorghum,74.55,83,No,Mobile Apps,691.11,Inorganic,Sprinkler,Yes,719.05
...,...,...,...,...,...,...,...,...,...,...,...,...
1988,Savannah Cultivators Club,Ghana,Wheat,97.13,29,Yes,Mobile Apps,555.20,Inorganic,Sprinkler,No,663.25
1993,Golden Agri Society,Kenya,Rice,90.02,80,Yes,Mobile Apps,664.49,NaN,Drip,Yes,710.92
1997,Prairie Cultivators Club,Ghana,Rice,89.37,43,No,Mobile Apps,603.02,Organic,Drip,No,958.79
1998,Sunrise Agricultural Association,Kenya,Wheat,87.01,99,No,Mobile Apps,370.42,Inorganic,Sprinkler,Yes,677.78


In [47]:
Digital_Agri_Pen['Irrigation'].unique()

array(['Sprinkler', 'Drip', nan], dtype=object)

In [48]:
Digital_Agri_Pen['Irrigation'].apply(type).value_counts()



Irrigation
<class 'str'>      421
<class 'float'>    229
Name: count, dtype: int64

In [49]:
Digital_Agri_Pen['Irrigation'].mode()[0]


'Sprinkler'

### Comment
The mode is 'Sprinkler'. Digital users (likely using Apps/SMS) prefer Sprinklers over Drip irrigation. This suggests a correlation between tech-savviness and a preference for automation (Sprinklers are easier to automate) over sheer efficiency (Drip is more efficient but maintenance-heavy).

## Q9: Large-Scale Technology Investment Patterns
**Business Question:** What is the average technology adoption level among large-scale farming operations (90+ acres), and how does it compare to smaller operations?

**Context:** Large farms typically have higher investment capacity and can be early adopters of expensive technologies.

In [50]:
Large_tech=Agri_tech[(Agri_tech['Farm_Size']>=90)]
Small_tech= Agri_tech[(Agri_tech['Farm_Size']<90)]
Large_tech

,Community_Name,Country,Crop_Type,Farm_Size,Tech_Adoption_Index,Drone_Usage,Mobile_Tech,Annual_Yield,Fertilizer_Type,Irrigation,Weather_Station,Annual_Rainfall
2,Prairie Agri Society,Nigeria,Wheat,97.46,14,Yes,SMS,439.41,NaN,Drip,Yes,1072.58
5,Savannah Agri Society,Tanzania,Barley,92.92,30,Yes,SMS,784.90,Organic,Sprinkler,No,872.48
12,Valley Agricultural Association,Ghana,Barley,99.69,60,No,Mobile Apps,182.56,Organic,Drip,Yes,1163.25
24,Meadow Farmers Group,Ghana,Rice,94.51,51,Yes,NaN,883.63,NaN,NaN,No,917.49
28,Valley Agricultural Association,Nigeria,Maize,98.92,1,No,NaN,571.98,NaN,Sprinkler,No,582.40
...,...,...,...,...,...,...,...,...,...,...,...,...
1936,Green Farmers Union,Tanzania,Barley,94.96,24,Yes,NaN,249.22,NaN,NaN,No,454.79
1947,Sunrise Farmers Group,Ghana,Barley,90.23,71,No,Mobile Apps,339.71,NaN,Sprinkler,No,580.10
1968,Harvest Growers Coalition,Tanzania,Barley,95.20,87,Yes,Mobile Apps,487.51,NaN,Sprinkler,Yes,583.40
1988,Savannah Cultivators Club,Ghana,Wheat,97.13,29,Yes,Mobile Apps,555.20,Inorganic,Sprinkler,No,663.25


In [51]:
print(Large_tech['Tech_Adoption_Index'].mean())

49.242009132420094


In [52]:
print(Small_tech['Tech_Adoption_Index'].mean())

50.14317798989332


### Comment
The tech adoption gap is non-existent (49.2 for Large vs. 50.1 for Small). This shatters the assumption that "Bigger = More Tech." In this dataset, small farms are just as modernized as giant ones, reinforcing the Q1 insight that Cooperatives (which aggregate small farmers) are effectively democratizing access to technology.


## Q10: Natural Farming Excellence
**Business Question:** What percentage of farming communities achieve high productivity (over 500 annual yield) without using fertilizers, demonstrating natural farming excellence?

**Context:** High-performing farms without fertilizer inputs represent efficient operations that could mentor others in sustainable practices.


In [53]:
Natural_farming=Agri_tech[(Agri_tech['Annual_Yield'] >= 500) & (Agri_tech['Fertilizer_Type'].isna())]
Natural_farming

,Community_Name,Country,Crop_Type,Farm_Size,Tech_Adoption_Index,Drone_Usage,Mobile_Tech,Annual_Yield,Fertilizer_Type,Irrigation,Weather_Station,Annual_Rainfall
3,Harvest Farmers Union,Nigeria,Maize,27.84,55,No,SMS,668.23,NaN,Sprinkler,No,629.73
4,Valley Farmers Group,Ghana,Sorghum,70.60,53,Yes,Mobile Apps,801.16,NaN,Sprinkler,No,909.69
18,Mountain Agri Society,Nigeria,Wheat,62.62,26,Yes,NaN,700.75,NaN,Sprinkler,Yes,1132.60
19,Valley Farmers Group,Kenya,Maize,33.14,43,No,Mobile Apps,943.97,NaN,NaN,No,589.20
20,Savannah Cultivators Club,Kenya,Sorghum,46.27,92,No,SMS,686.81,NaN,NaN,Yes,612.64
...,...,...,...,...,...,...,...,...,...,...,...,...
1983,Golden Agri Society,South Africa,Sorghum,89.55,99,No,NaN,701.67,NaN,Drip,No,526.19
1984,Meadow Farmers Union,Ghana,Wheat,84.12,39,No,SMS,845.32,NaN,Drip,No,901.27
1989,Harvest Agri Society,Kenya,Rice,55.40,24,No,SMS,658.68,NaN,Sprinkler,Yes,693.57
1990,Golden Farmers Union,Nigeria,Wheat,19.07,6,No,SMS,904.24,NaN,NaN,No,736.37


In [54]:
Natural_farming['Community_Name']

3           Harvest Farmers Union
4            Valley Farmers Group
18          Mountain Agri Society
19           Valley Farmers Group
20      Savannah Cultivators Club
                  ...            
1983          Golden Agri Society
1984         Meadow Farmers Union
1989         Harvest Agri Society
1990         Golden Farmers Union
1993          Golden Agri Society
Name: Community_Name, Length: 374, dtype: object

In [55]:
Agri_tech[(Agri_tech['Annual_Yield'] >= 500) & (Agri_tech['Fertilizer_Type'].isna())]

,Community_Name,Country,Crop_Type,Farm_Size,Tech_Adoption_Index,Drone_Usage,Mobile_Tech,Annual_Yield,Fertilizer_Type,Irrigation,Weather_Station,Annual_Rainfall
3,Harvest Farmers Union,Nigeria,Maize,27.84,55,No,SMS,668.23,NaN,Sprinkler,No,629.73
4,Valley Farmers Group,Ghana,Sorghum,70.60,53,Yes,Mobile Apps,801.16,NaN,Sprinkler,No,909.69
18,Mountain Agri Society,Nigeria,Wheat,62.62,26,Yes,NaN,700.75,NaN,Sprinkler,Yes,1132.60
19,Valley Farmers Group,Kenya,Maize,33.14,43,No,Mobile Apps,943.97,NaN,NaN,No,589.20
20,Savannah Cultivators Club,Kenya,Sorghum,46.27,92,No,SMS,686.81,NaN,NaN,Yes,612.64
...,...,...,...,...,...,...,...,...,...,...,...,...
1983,Golden Agri Society,South Africa,Sorghum,89.55,99,No,NaN,701.67,NaN,Drip,No,526.19
1984,Meadow Farmers Union,Ghana,Wheat,84.12,39,No,SMS,845.32,NaN,Drip,No,901.27
1989,Harvest Agri Society,Kenya,Rice,55.40,24,No,SMS,658.68,NaN,Sprinkler,Yes,693.57
1990,Golden Farmers Union,Nigeria,Wheat,19.07,6,No,SMS,904.24,NaN,NaN,No,736.37


In [56]:
All_Farming = Agri_tech.shape[0]
Natural_farming_count = Natural_farming.shape[0] 
Percentage_Farm = (Natural_farming_count/ All_Farming) * 100 

Percentage_Farm

18.7

### Comment
Nearly 1 in 5 high-performing farms (18.7%) use no recorded fertilizer. This proves that "Natural Farming" isn't just a niche or subsistence method—it is a scalable, high-output production model capable of competing directly with conventional chemical farming in this region.